src/message/from_prior/pack.rs

```diff
@@ -0,0 +1,225 @@
+use crate::{
+    did::DIDResolver,
+    error::{err_msg, ErrorKind, Result, ResultContext, ResultExt},
+    jws::{self, Algorithm},
+    message::from_prior::JWT_TYP,
+    secrets::SecretsResolver,
+    utils::{
+        crypto::{AsKnownKeyPair, KnownKeyPair},
+        did::did_or_url,
+    },
+    FromPrior,
+};
+
+impl FromPrior {
+    pub async fn pack<'dr, 'sr>(
+        &self,
+        issuer_kid: Option<&str>,
+        did_resolver: &'dr (dyn DIDResolver + 'dr),
+        secrets_resolver: &'sr (dyn SecretsResolver + 'sr),
+    ) -> Result<(String, String)> {
+        let from_prior_str = serde_json::to_string(self)
+            .kind(ErrorKind::InvalidState, "Unable serialize message")?;
+
+        let did_doc = did_resolver
+            .resolve(&self.iss)
+            .await
+            .context("Unable to resolve from_prior issuer DID")?
+            .ok_or_else(|| {
+                err_msg(
+                    ErrorKind::DIDNotResolved,
+                    "from_prior issuer DIDDoc is not found",
+                )
+            })?;
+
+        let authentication_kids: Vec<&str> = if let Some(v) = issuer_kid {
```

Comment:
I would suggest simplify naming by avoid intermediate names

```rust
ids: Vec<&str> = if let Some(issuer_kid) = issuer_kid {
             let (did, kid) = did_or_url(issuer_kid);

             let kid = kid.ok_or_else(|| {
                 err_msg(ErrorKind::Malformed, "issuer_kid content is not DID URL")
             })?;
```

Also as we work only in one context we can avoid giving context in names. For example, `authentication_kids` can be just `kids` here.

Reply:
Renamed `v` to `issuer_kid`.